extensions/wikia/PortableInfobox/tests/PortableInfoboxParserTagControllerTest.php

```diff
@@ -51,97 +53,50 @@ public function testEmptyInfobox() {
 		$text = '<data><default>test</default></data>';
-		$themeVariableName = 'variableName';
-		$themeName = 'variable';
-		$defaultTheme = 'default';
 
 		$marker = $this->controller->renderInfobox( $text,
-			[ 'theme' => $defaultTheme, 'theme-source' => $themeVariableName ],
+			[ 'theme' => $staticTheme, 'theme-source' => 'testVar' ],
 			$this->parser,
-			$this->parser->getPreprocessor()->newFrame() )[ 0 ];
-		$output = $this->controller->replaceMarkers( $marker );
-
-		$this->assertTrue( $this->checkClassName(
-			$output,
-			PortableInfoboxParserTagController::INFOBOX_THEME_PREFIX . $defaultTheme
-		) );
-	}
-
-	public function testNoThemeInfobox() {
-		$text = '<data><default>test</default></data>';
-
-		$marker = $this->controller->renderInfobox( $text, [ ], $this->parser,
-			$this->parser->getPreprocessor()->newFrame() )[ 0 ];
-		$output = $this->controller->replaceMarkers( $marker );
-
-		$this->assertTrue( $this->checkClassName(
-			$output,
-			PortableInfoboxParserTagController::INFOBOX_THEME_PREFIX . PortableInfoboxParserTagController::DEFAULT_THEME_NAME
-		) );
-	}
-
-	public function testWhiteSpacedThemeInfobox() {
-		$text = '<data><default>test</default></data>';
-		$defaultTheme = 'test test';
-		$expectedName = 'test-test';
-
-		$marker = $this->controller->renderInfobox( $text, [ 'theme' => $defaultTheme ], $this->parser,
-			$this->parser->getPreprocessor()->newFrame() )[ 0 ];
+			$this->parser->getPreprocessor()->newCustomFrame( [ 'testVar' => $variableTheme ] ) )[ 0 ];
 		$output = $this->controller->replaceMarkers( $marker );
 
-		$this->assertTrue( $this->checkClassName(
-			$output,
-			PortableInfoboxParserTagController::INFOBOX_THEME_PREFIX . $expectedName
-		) );
+		$this->assertTrue( array_reduce( $classes, function ( $result, $class ) use ( $output ) {
+			return $result && $this->containsClassName( $output, $class );
+		}, true ), $message );
 	}
 
-	public function testMultiWhiteSpacedThemeInfobox() {
-		$text = '<data><default>test</default></data>';
-		$defaultTheme = "test    test\n test\ttest";
-		$expectedName = 'test-test-test-test';
-
-		$marker = $this->controller->renderInfobox( $text, [ 'theme' => $defaultTheme ], $this->parser,
-			$this->parser->getPreprocessor()->newFrame() )[ 0 ];
-		$output = $this->controller->replaceMarkers( $marker );
-
-		$this->assertTrue( $this->checkClassName(
-			$output,
-			PortableInfoboxParserTagController::INFOBOX_THEME_PREFIX . $expectedName
-		) );
+	public function themeNamesProvider() {
+		return [
+			// static theme, variable name, variable theme, [ classes ], message
+			[ ' ', '', [
+				PortableInfoboxParserTagController::INFOBOX_THEME_PREFIX . PortableInfoboxParserTagController::DEFAULT_THEME_NAME
+			], "Should use default when theme names are invalid" ],
+			[ 'test', null, [ PortableInfoboxParserTagController::INFOBOX_THEME_PREFIX . 'test' ],
+				"Should contain static theme" ],
+			[ null, 'variable', [
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "variable" 4 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1192)